django/db/models/query.py

```diff
@@ -1566,12 +1566,13 @@ def prefetch_one_level(instances, prefetcher, lookup, level):
     #  callable that gets value to be matched for returned instances,
     #  callable that gets value to be matched for passed in instances,
     #  boolean that is True for singly related objects,
-    #  cache name to assign to).
+    #  cache or field name to assign to,
+    #  boolean that is True when the previous argument is a cache name vs a field name).
 
     # The 'values to be matched' must be hashable as they will be used
     # in a dictionary.
 
-    rel_qs, rel_obj_attr, instance_attr, single, cache_name = (
+    rel_qs, rel_obj_attr, instance_attr, single, cache_name, is_cache = (
```

Comment:
Hmm. I'm not sure about `is_field` -- don't we generally call them all fields, including FKs etc?

I'm not coming up with anything better than `is_descriptor`. I can see why it feels too "implementation specific", but on the other hand this flag only exists deep inside the implementation of prefetch related. And I think `is_descriptor` accurately captures the conditions under which we need to just set the attribute rather than populating the cache, doesn't it? Can you think of a likely case (even a hypothetical future one) where that term would become too specific?